### ISSUE TITLE
VEATIC dataset processing

### ISSUE THREAD

**Issue by DesmondChoy** [keimiii/moodsic#12]
# Overview

VEATIC is a video-based affect dataset with per-frame continuous valence–arousal (V/A) annotations for a selected character, collected via real-time annotation across 124 clips from movies, documentaries, and home videos. It emphasizes context beyond facial crops and provides a simple baseline and evaluation metrics (CCC/PCC/RMSE/SAGR).

## Links 

- Project page: https://veatic.github.io/
- arXiv: https://arxiv.org/abs/2309.06745
- WACV 2024 (OpenAccess): https://openaccess.thecvf.com/content/WACV2024/html/Ren_VEATIC_Video-Based_Emotion_and_Affect_Tracking_in_Context_Dataset_WACV_2024_paper.html
- Code: https://github.com/AlbusPeter/VEATIC

Notes:

- 124 annotated videos; videos 98–123 include interacting characters with separate per-character ratings.
- Labels are frame-by-frame continuous V/A; psychophysics per-rater series available for 0–82.
- Dataset structure: video/{id}.mp4, rating_averaged/{id}_{valence,arousal}.csv, optionally frames/{id}/{frame}.jpg.
- Intended use includes benchmarking models with context + character cues.

# Pain Point

Using raw per-frame V/A as targets/ground truth is too noisy and mismatched to affect dynamics (typically evolve over hundreds of ms to seconds). We need lightweight temporal preprocessing so labels better reflect perceptual timescales and model output cadence.

# Proposal (POC‑friendly defaults)

Temporal smoothing of labels: 

> Think of it like a gentle “motion blur” over time. A single frame can twitch or flicker, but moods shift over seconds. Blending about one second of labels turns spiky frame-by-frame noise into a smoother curve that better matches how emotions actually unfold.

- Apply a simple Gaussian low-pass over the per-frame V/A series for each video.
- Default: window ≈ 1.0 s, sigma ≈ 0.25–0.4 s (e.g., at 25 fps: window 25 frames, sigma 6 frames).
- One-liner with gaussian_filter1d; robust and predictable; materially reduces jitter with negligible implementation cost.

Annotator delay compensation: 

> Humans react a beat late when sliding a control in real time. We nudge the label curve slightly earlier (about 0.7 seconds) so it lines up with what’s on screen, like fixing lip‑sync so the voice matches the mouth.

- Shift smoothed labels backward by a fixed 0.7 s to counter human reaction lag in continuous annotation.
- Default: 0.7 s (clip at edges, no wrap).
- Constant shift is trivial to implement and commonly improves CCC in continuous affect tasks; no tuning required to get value.

Downsample labels after smoothing: 

> We don’t need 25 updates per second for emotions. Reporting 5 times per second keeps the important changes while cutting jitter and compute. It’s like checking the weather hourly instead of every minute—you still see the trend without the noise.

- Produce 5 Hz targets to match typical model head cadence and reduce noise further.
- Default: average-pool windows to 5 Hz (at 25 fps: pool every 5 frames).
- Simple aggregation; reduces compute and label variance; aligns with model timescales without changing architecture.

Artifact handling and provenance: 

> We save these cleaned-up labels next to the raw ones with a simple note about what we did (how much smoothing, shift, new rate). It’s non‑destructive—like editing a photo without overwriting the original—so we can compare or roll back anytime.

- Write derived labels alongside originals with metadata.
- Default: save to rating_smoothed/{id}_{valence,arousal}.csv including header metadata (fps, filter, sigma/window, delay, output_fps).
- Cheap bookkeeping; preserves reproducibility and lets us toggle between raw/smoothed easily.

**Comment by jrlyy** [comment]
with vs without temporal smoothing 

<img width="1434" height="691" alt="Image" src="https://github.com/user-attachments/assets/2a90c5fb-a510-4313-998b-c463049c39b0" />
<img width="1339" height="662" alt="Image" src="https://github.com/user-attachments/assets/e7d964d8-a562-4558-bced-91651d7634e5" />

**Comment by jrlyy** [comment]
The 4 videos we can use as inference for each quadrant

Quadrant 1 (+V/+A): 84 (99.2%)
Quadrant 2 (+V/-A): 31 (77.8%)
Quadrant 3 (-V/+A): 23 (99.7%)
Quadrant 4 (-V/-A): 37 (99.2%)

**Comment by jrlyy** [comment]
Video 84 
<img width="926" height="126" alt="Image" src="https://github.com/user-attachments/assets/add5ef7e-de06-40e2-848d-1decaba425cc" />

Pre-trained model: 

> Video info: 876 frames, 25.00 fps
> Extracting frames at indices: [  0 218 437 656 875]
> Extracted 5 frames
> Input tensor shape: torch.Size([1, 5, 3, 640, 480])
> 
> Prediction Results:
> Valence: 0.8851
> Arousal: 0.1612
> Frames used: 5
> Raw predictions: [1.3987068  0.16256821]

Clip model: Does the huge drop in valence is because it seems another target face is not in the scene, it's another character. Don't know how to explain the arousal


Metrics for Video 84:
Valence - MSE: 0.2328, MAE: 0.4074
Arousal - MSE: 0.4003, MAE: 0.6180

Average V-A Scores for 5 Selected Frames of Video 84:
Selected Frame Indices: [0, 218, 437, 656, 875]
Predicted - Valence: 0.1288, Arousal: -0.3781
Ground Truth - Valence: 0.3820, Arousal: 0.2399
Difference - Valence: 0.2532, Arousal: 0.6180

Individual Frame Values:
Frame 0: Pred V=-0.1579, GT V=0.0158, Pred A=-0.5388, GT A=0.0002
Frame 218: Pred V=0.1052, GT V=0.5353, Pred A=-0.4787, GT A=0.2556
Frame 437: Pred V=0.4074, GT V=0.5764, Pred A=-0.2215, GT A=0.3527
Frame 656: Pred V=-0.2976, GT V=0.5809, Pred A=-0.3729, GT A=0.4357
Frame 875: Pred V=0.5871, GT V=0.2016, Pred A=-0.2788, GT A=0.1552

<img width="1143" height="549" alt="Image" src="https://github.com/user-attachments/assets/2c4bcd55-4cf2-44d0-b680-0d31a9ae0e84" />

**Comment by jrlyy** [comment]
Video 94 
[0  515 1031 1547 2063] 
<img width="1990" height="257" alt="Image" src="https://github.com/user-attachments/assets/ed2a78e9-178a-4c25-b903-16de72347a66" />

<img width="1366" height="715" alt="Image" src="https://github.com/user-attachments/assets/6e465331-0c54-4de6-b567-28418e9cf64b" />

Ground truth average :
> Valence: 0.4154
> Arousal: 0.3661

Clip model average :
 > Valence: 0.1695
> Arousal: -0.4025

Pre-trained model:
> Valence: 0.6569
> Arousal: -0.0710
> Frames used: 5
> Raw predictions: [ 0.7873908  -0.07109426]

<img width="1379" height="754" alt="Image" src="https://github.com/user-attachments/assets/23aab94f-ddc5-4e0e-830b-b969cd4e5890" />

**Comment by jrlyy** [comment]
Reopen because I edited some videos and need to get the exact frames to cut instead of seconds

**Comment by jrlyy** [comment]
============================================================
VIDEO EMOTION EVALUATION (Video ID: 23)
============================================================

Aggregation Method: mean
Frame Count Processed: 15

Ground Truth:
  Valence:  -0.588
  Arousal:  +0.442
  Quadrant: -+

Prediction:
  Valence:  -0.196
  Arousal:  +0.359
  Quadrant: -+

Result: ✓ SUCCESS
Euclidean Distance: 0.400
============================================================

Recommended Song:
  Title:    Heat Index
  ID:       60
  Cluster:  3
  V/A:      (-0.175, +0.350)
  Genre:    blues/soul

**Comment by jrlyy** [comment]
**VIDEO EMOTION EVALUATION (Video ID: 94)**

Aggregation Method: mean
Frame Count Processed: 21

Ground Truth:
  Valence:  +0.606
  Arousal:  +0.510
  Quadrant: ++

Prediction:
  Valence:  +0.186
  Arousal:  +0.243
  Quadrant: ++

**Result: ✓ SUCCESS
Euclidean Distance: 0.498**

Recommended Song:
  Title:    Tleilaxuburg5 Movement II
  ID:       1159
  Cluster:  2
  V/A:      (+0.200, +0.250)
  Genre:    electronic

**Comment by jrlyy** [comment]
**VIDEO EMOTION EVALUATION (Video ID: 31)**

Aggregation Method: mean
Frame Count Processed: 13

Ground Truth:
  Valence:  +0.120
  Arousal:  -0.125
  Quadrant: +-

Prediction:
  Valence:  -0.154
  Arousal:  +0.171
  Quadrant: -+

**Result: ✗ FAIL
Euclidean Distance: 0.404**

Recommended Song:
  Title:    AcidJazz
  ID:       658
  Cluster:  3
  V/A:      (-0.150, +0.175)
  Genre:    jazz
